src/include/migraphx/raw_data.hpp

```diff
@@ -125,7 +125,7 @@ struct raw_data : raw_data_base
         {
             auto&& buffer    = static_cast<const Derived&>(*this).data();
             shape view_shape = {shape::uint8_type, {s.bytes()}};
-            v(make_view(view_shape, reinterpret_cast<byte*>(buffer)));
+            v(make_view(view_shape, const_cast<byte*>(reinterpret_cast<const byte*>(buffer))));
```

Comment:
Dont use `const_cast`. The type should be `const byte*` when `data()` is a `const char*`.

Reply:
Probably something like `using type = std::conditional_t<std::is_const<decltype(buffer)>{}, const byte*, byte*>`

Reply:
It's doable but surprisingly ugly: 
```
using byte_type = std::conditional_t<std::is_const_v<std::remove_pointer_t<std::remove_reference_t<decltype(buffer)>>>, const byte*, byte*>;
```
Updating